client/my-sites/stats/stats-purchase/stats-commercial-upgrade-slider/index.tsx

```diff
@@ -106,6 +110,16 @@ function StatsCommercialUpgradeSlider( {
 	const steps = getStepsForTiers( tiers );
 
 	const handleSliderChanged = ( index: number ) => {
+		recordTracksEvent(
+			`${
+				isOdysseyStats ? 'jetpack_odyssey' : 'calypso'
+			}_stats_purchase_commercial_slider_clicked`,
+			{
+				tier_views: tiers[ index ]?.views,
```

Comment:
Would passing an indicator aligned with [the PWYW slider](https://github.com/Automattic/wp-calypso/blob/8bf44a682056cd95b2b19dc07e34f5f6ece3f368/client/my-sites/stats/stats-purchase/stats-pwyw-uprade-slider/index.tsx#L125) for similar track events be great to make the analytics more comparable? Or, do we just need to analyze the final purchases rather than the slider-changed behaviors?

Reply:
I think tracking how user is interacting with it makes sense. However, I would recommend to do it outside the component, which we are probably going to share with other products.

Reply:
This is happening outside `<TierUpgradeSlider />` and I think we would share that component. Is that what you had in mind @kangzj ?

Reply:
@dognose24 when users are upgrading we don't show the tier they already own so it's possible that the same index will correspond to different view limits.

Reply:
Correct me if I'm wrong @a8ck3n, I vaguely remember Kevin mentioned we'd like to share `client/my-sites/stats/stats-purchase/stats-commercial-upgrade-slider` and `client/my-sites/stats/stats-purchase/stats-commercial-upgrade-slider` too probably with AI, and that's the reason he did a refactoring, and isolated them. 🤔

Reply:
Ok, I'll move it up.

Reply:
Actually, I can't move it up easily because the PWYW slider calculates the initial values in the slider wrapper-wrapper and the display initial value in the slider wrapper. I need that information to add a parameter to the tracking event.
I could pass those values up alongside the selection, but I don't think it will be good to do it just for tracking. I'll extend the component with a prop enabling tracking and another prop accepting custom event names.

Reply:
> so it's possible that the same index will correspond to different view limits.

Yes, so we might need to avoid using a comprehensive indicator, e.g., [step for index](https://github.com/Automattic/wp-calypso/blob/8bf44a682056cd95b2b19dc07e34f5f6ece3f368/client/my-sites/stats/stats-purchase/stats-pwyw-uprade-slider/index.tsx#L125), which would not be so meaningful when analyzing afterward.

Reply:
> share client/my-sites/stats/stats-purchase/stats-commercial-upgrade-slider and client/my-sites/stats/stats-purchase/stats-commercial-upgrade-slider too probably with AI

@kangzj There are two `client/my-sites/stats/stats-purchase/stats-commercial-upgrade-slider`. 😄

Reply:
> so it's possible that the same index will correspond to different view limits.

@grzegorz-cp Yes, so we might need to avoid using a comprehensive indicator, e.g., [step for index](https://github.com/Automattic/wp-calypso/blob/8bf44a682056cd95b2b19dc07e34f5f6ece3f368/client/my-sites/stats/stats-purchase/stats-pwyw-uprade-slider/index.tsx#L125), which would not be so meaningful when analyzing afterward.

Reply:
I don't follow that comment @dognose24. We need to use an index for PWYW because different currencies have different values and we have to use a limit for the tiers because the same index doesn't mean the same tier if someone upgrades.

Reply:
Hello!

> I vaguely remember Kevin mentioned we'd like to share

Actually, my thinking was that the TierUpgradeSlider component be the "dumb" component which is shared while the two wrappers (commercial and pwyw) are the context-specific wrappers. With that in mind, I think this is the correct place for any tracking logic.